### ISSUE TITLE
Drop Down Filters Not Listing Available Options - Errors in Log

### ISSUE THREAD

**Issue by slickrock22** [metabase/metabase#14070]
**Describe the bug**
An existing dashboard filters have reverted to “Plain Input Box”,  When filtering on this field is set to “A List of All Values”.  I've encountered this situation before and sometimes they've had to toggle the selection between two different types, discard and rescan as it eventually shows back up on the dashboard. I have tried that in this case and have not been successful.

**Logs**
Please include javascript console and server logs around the time this bug occurred. For information about how to get these,  :error {"3db2f415" {:name (not ("Non-blank string" ""))}},
 :via
 [{:type clojure.lang.ExceptionInfo,
   :message "Output of dashboard->resolved-params does not match schema: \n\n\t   {\"3db2f415\" {:name (not (\"Non-blank string\" \"\"))}}  \n\n", 

There seems to be some issue in the logs that are not allowing any of the filters on the dashboard to operate as a drop down list.

**To Reproduce**
Not sure hot to reproduce in another environment.   I tested a few different scenarios. I first went in and added a new filter to the existing dashboard to see whether or not when I created that filter and applied it to a question, whether or not it would show values in the list. It does not. I've been cloned or duplicated the dashboard to see whether or not the filters would work. It does not are then created a new dashboard from scratch and added one of the questions from the existing dashboard and added a filter to the new dashboard and assigned it to that question. When I clicked on that new filter it did show the drop down list values.  I have read about filters not displaying when results exceed a certain count.  Some of these filters have 5-10 options only.

**Information about your Metabase Installation:**
I am running v37.3 in ELB with Postgres backend.

{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.7+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.7",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.7+10",
    "os.name": "Linux",
    "os.version": "4.14.181-108.257.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "sqlserver"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "11.8"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-12-03",
      "tag": "v0.37.3",
      "branch": "release-x.37.x",
      "hash": "2f1e783"
    },
    "settings": {
      "report-timezone": "US/Eastern"
    }
  }
}
**Severity**
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
Severe.  Dashboard that are embedded are not functional.

**Comment by flamber** [comment]
Basically a duplicate of #13537+#13832 (perhaps #12654), where dropdown information is not shown for unauthenticated users.

**Comment by slickrock22** [comment]
@flamber I just fixed this issue in all of my dashboards by identifying the offending value in the db. In my case, every dashboard had {"name":"","slug":"","id":"3db2f415","type":"date/all-options"} as the first value in the parameters field.

SELECT * FROM public.report_dashboard WHERE parameters like '%"name":"","slug"%'

You simply delete the value from the beginning of the string, save and my dashboards all worked. Not sure if the value will be the same for everyone but hopefully this provides some direction. It looked odd becuase it didnt reference a real filter. Also I noticed in the UI, when I looked at the embedding page, I have a filter value for embedding that was blank.
